server/utils.go

```diff
@@ -142,3 +148,49 @@ func weiBigIntToEthBigFloat(wei *big.Int) (ethValue *big.Float) {
 	ethValue = new(big.Float).Quo(fbalance, big.NewFloat(1e18))
 	return
 }
+
+func ComputeBlockHash(payload *bellatrix.ExecutionPayload) (phase0.Hash32, error) {
+	header, err := executionPayloadToBlockHeader(payload)
+	if err != nil {
+		return phase0.Hash32{}, err
+	}
+	return phase0.Hash32(header.Hash()), nil
+}
+
+func executionPayloadToBlockHeader(payload *bellatrix.ExecutionPayload) (*types.Header, error) {
+	transactionData := make([]*types.Transaction, len(payload.Transactions))
+	for i, encTx := range payload.Transactions {
+		var tx types.Transaction
+		err := tx.UnmarshalBinary(encTx)
+		if err != nil {
```

Comment:
Just a little styling comment.

```suggestion

		if err := tx.UnmarshalBinary(encTx); err != nil {
```